### PR TITLE
Add docs regarding config --list

### PR DESCRIPTION
**Pull request by isidentical**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Review by efiop — APPROVED** [review]
Not sure if we want a special paragraph in `description` section for `--list` CC @jorgeorpinel . LGTM.

File: content/docs/command-reference/config.md [hunk 1]
```diff
@@ -5,8 +5,8 @@ Get or set <abbr>project</abbr>-level (or global) DVC configuration options.
 ## Synopsis
 
 ```usage
-usage: dvc config [-h] [--global | --system | --local] [-q | -v] [-u]
-                  name [value]
+usage: dvc config [-h] [--global | --system | --local] [-q | -v] [-u] [-l]
+                  [name] [value]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/config.md:9]:
```suggestion
usage: dvc config [-h] [--global | --system | --local] [-q | -v] [-u]
                  [-l] [name] [value]
```

**Comment by jorgeorpinel** [comment]
A sentence in the paragraph starting with "If the config option value is not provided" could be useful but not required.

**Review by jorgeorpinel — APPROVED** [review]
Another good option would be to add an example about this to have a sample of the output of `--list`, but oh well.

**Review by shcheklein — APPROVED** [review]

**Pull request merged**